### ISSUE TITLE
unhandled exception while running flutter doctor command 

### ISSUE THREAD

**Issue by jrsadik7** [flutter/flutter#8085]
```
root@jafferpc:/home/jaffer/Jaffar/Android/flutter# flutter doctor
Unhandled exception:
RangeError: Value not in range: -1
#0      _StringBase.substring (dart:core-patch/string_patch.dart:366)
#1      getPlatformLocale (package:usage/src/usage_impl_io.dart:145)
#2      _createUserAgent (package:usage/src/usage_impl_io.dart:44)
#3      IOPostHandler.IOPostHandler (package:usage/src/usage_impl_io.dart:80)
#4      AnalyticsIO.AnalyticsIO (package:usage/src/usage_impl_io.dart:32)
#5      Usage.Usage (package:flutter_tools/src/usage.dart:24)
#6      main.<main_async_body>.<anonymous closure>.<<anonymous closure>_async_body>.<anonymous closure> (package:flutter_tools/executable.dart:122)
#7      AppContext.putIfAbsent (package:flutter_tools/src/base/context.dart:46)
#8      main.<main_async_body>.<anonymous closure>.<<anonymous closure>_async_body> (package:flutter_tools/executable.dart:122)
#9      Future.Future.microtask.<anonymous closure> (dart:async/future.dart:184)
#10     _rootRun (dart:async/zone.dart:1150)
#11     _CustomZone.run (dart:async/zone.dart:1026)
#12     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:953)
#13     _microtaskLoop (dart:async/schedule_microtask.dart:41)
#14     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
#15     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#16     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149)
root@jafferpc:/home/jaffer/Jaffar/Android/flutter# 
```

**Comment by jrsadik7** [comment]
# OS
Ubuntu 14.0.4

# JAVA
 java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)

# Intellij Community Edition
   *Dart plugin : intalled
   *flutter plugin: installed
   *flutter sdk : clonned


can anybody please help me .....

thanks in Advance :)

**Comment by eseidelGoogle** [comment]
This looks like a regression from the latest package:usage update?  @devoncarew (fyi @tvolkert who has been working in other areas of tool improvement recently).

**Comment by jrsadik7** [comment]
can anyone please guide me to resolve this issue

**Comment by tvolkert** [comment]
@jrsadik7 once https://github.com/flutter/flutter/pull/8130 is merged into master, you should be able to `git pull`, then `flutter update-packages --upgrade`, and the issue should be fixed.

**Comment by jrsadik7** [comment]
**resolved** 

I deleted old flutter sdk 
cloned new sdk from git
configured flutter PATH

######################################################################## 100.0%
Building flutter tool...

  ╔════════════════════════════════════════════════════════════════════════════╗
  ║                 Welcome to Flutter! - https://flutter.io                   ║
  ║                                                                            ║
  ║ The Flutter tool anonymously reports feature usage statistics and basic    ║
  ║ crash reports to Google in order to help Google contribute improvements to ║
  ║ Flutter over time. See Google's privacy policy:                            ║
  ║ https://www.google.com/intl/en/policies/privacy/                           ║
  ║                                                                            ║
  ║ Use "flutter config --no-analytics" to disable analytics reporting.        ║
  ╚════════════════════════════════════════════════════════════════════════════╝
  
Downloading Material fonts...                         1.9s
Downloading package sky_engine...                     1.5s
Downloading linux-x64 tools...                       13.8s
Downloading android-arm-profile/linux-x64 tools...   26.5s
Downloading android-arm-release/linux-x64 tools...   15.1s
Downloading android-x86 tools...                     13.9s
Downloading android-x64 tools...                     11.9s
Downloading android-arm tools...                     11.3s
Downloading android-arm-profile tools...             13.1s
Downloading android-arm-release tools...              8.3s
[✓] Flutter (on Linux, channel master)
    • Flutter at /home/jaffer/Jaffar/Android/flutter
    • Framework revision 948ca22923 (18 minutes ago), 
    • Engine revision f306a6f701
    • Tools Dart version 1.22.0-dev.10.6

[✗] Android toolchain - develop for Android devices
    x Android Studio / Android SDK not found. Download from https://developer.android.com/sdk/
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).

[✗] Flutter IDE Support (No supported IDEs installed)
    • IntelliJ - https://www.jetbrains.com/idea/

[✓] Connected devices
    • None


thanks for the update and fix :+1:
